### ISSUE TITLE
Reko works incorrectly with floating point compare statement

### ISSUE THREAD

**Issue by ptomin** [uxmal/reko#348]
For instance, it could be reproduced on `VCExeSample`. `(d < 321.0)` statement was decompiled as `globals->r4020F0 < rArg04` (see https://github.com/uxmal/reko/commit/37ab72ddc33da38e49a179646f5944d056083b58#commitcomment-19666936 and https://github.com/uxmal/reko/commit/37ab72ddc33da38e49a179646f5944d056083b58#commitcomment-19666961) but it should be `globals->r4020F0 > rArg04`

**Comment by uxmal** [comment]
The `d > 1.0` in `test7` is also inverted. Likely a bug in the code that deals with
```asm
fld1	
	fcomp double ptr [ebp+08]
	fstsw	 ax
	test ah,05
	jpe 00401189
```

**Comment by uxmal** [comment]
The culprit is likely in the `MatchesFstswSequence()`  method in `https://github.com/uxmal/reko/blob/master/src/Arch/X86/X86Rewriter.Fpu.cs`. Pavel, do you want to take a stab at this?

**Comment by ptomin** [comment]
Yes, I could try to fix it

**Comment by ptomin** [comment]
Fixed in a20e4fd. Could be closed, now
